### DISCUSSION TITLE
Support for git-tags datasource with Poetry manager

### DISCUSSION THREAD

**Discussion by sandy-wright** [renovatebot/renovate#18295] [Suggest an Idea]
Hi there! First time posting, so apologies if I commit any kind of faux-pas or whatnot - feel free to point it out if I do ☺️ 

I'm curious about why the `git-tags` data source isn't supported with the Poetry manager, and I'm wondering if there's any interest in adding the feature.

The details of my situation in particular:

* **Renovate config:** Self-hosted using a Docker image running in GitHub Actions
* **Renovate version:** 32.213.2 (non-slim base image)
* **Poetry version:** 1.2.0
* **Python version:** >3.8, <3.11

My org publishes internal packages as releases in GitHub with tags. We do not have a private PyPI data source (nor is there any interest in doing so - we're too small to really benefit from that).

As such, we have Poetry deps defined as follows:
```toml
org_lib = {git = "git+ssh://git@github.com/<org>/<org-lib>.git", tag = "v#.#.#"}
```

I have configured Renovate with all the SSH creds it needs, so it can access this repository no problem. Poetry also has no complaints when it resolves the dependencies.

However, I can't use Renovate to update these dependencies because it's always skipped:
```
{
   "depName": "org_lib",
   "depType": "dependencies",
   "currentValue": "",
   "managerData": {"nestedVersion": false},
   "datasource": "pypi",
   "lockedVersion": "#.#.#",
   "skipReason": "git-dependency",
   "depIndex": #,
   "updates": []
 },
 ```
 
We use microservices, so there's a lot of `pyproject.toml` files pointing at these deps. Being able to use Renovate to update them would be _amaaaazing_.
 
So, my main questions:

1. Is there a reason that this functionality doesn't exist with the Poetry manager at the moment? I saw that it was implemented this way in the original PR, but I'm not sure _why_.
2. Is there a work-around to get this done by configuring other data sources for Poetry in the `renovate.json` file, perhaps?
3. Would this be a reasonable feature request and/or is this something that a new contributor (like me! 😄) could potentially add to the codebase?

Please let me know if you need any more info from me - thanks!!

Extra context:
* [Original Poetry PR](https://github.com/renovatebot/renovate/pull/3440)
* [Source code for the issue](https://github.com/renovatebot/renovate/blob/main/lib/modules/manager/poetry/extract.ts#L64-L75)
* [Maybe related issue for bundler?](https://github.com/renovatebot/renovate/issues/4789)

**Comment by rarkins** [comment]
Hi @sandy-wright, welcome to the repo!

I'm pretty sure the answer is just that nobody's implemented it yet, so therefore you would be welcome to.

Hopefully, if you just add the logic to poetry's extract function, and the `git-tags` reference is correct constructed, then that should be all that's required.

BTW if you're using `github.com` you may want to use `github-tags` instead of `git-tags` as it can make credential handling easier/automatic.

But possibly, you might find that:
- Renovate updates the package file fine, but
- Running `poetry` fails because it does not have permissions to access the private github repo

Do you know how Renovate should pass a GitHub token to Poetry if so?

There is already some logic for credentials in the Poetry artifacts updating: https://github.com/renovatebot/renovate/blob/e6c43aae81fc56b87137904a51d574cd26f9b607/lib/modules/manager/poetry/artifacts.ts#L105-L110

So if you'd like to go ahead with this:
- Set up Renovate to run locally (so you can `yarn run abc/def` against your preferred repo
- Make sure your github token has access to not only the repo you run against but any repos you want to fetch tags from
- Improve the poetry extract function to include git or github tags deps
- Verify that Renovate can successfully look up and make updates to the pyproject.yaml
- Now verify if the lock file updating subsequently fails
- If so, work out what credentials need to be passed to Poetry and update the existing credentials logic already present
